aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h

```diff
@@ -368,6 +379,17 @@ namespace impl {
     }
   };
 
+  template<class T, bool AllowDeprecatedTypes>
+  struct ivalue_to_arg<OptionalArrayRef<T>, AllowDeprecatedTypes> final {
+    // If an argument is optional<ArrayRef<T>>, convert the IValue to an
+    // optional<std::vector<T>> and pass that to the operator. OptionalArray<T>
+    // is basically a optional<std::vector<T>> but impliticly convertible to
+    // OptionalArrayRef<T>
+    static OptionalArrayRef<T> call(IValue& v) {
```

Comment:
This returns a reference to the temporary `OptionalArray<T>`, it should be returning it directly
```suggestion
    static OptionalArray<T> call(IValue& v) {
```

Reply:
Ooh, I didn't realize `ivalue_to_arg<optional<ArrayRef<T>>::call()` was returning `OptionalArray<T>`. This will probably fix the issue, thanks!